extensions/ql-vscode/src/run-remote-query.ts

```diff
@@ -413,3 +420,105 @@ export async function attemptRerun(
     void showAndLogErrorMessage(error);
   }
 }
+
+interface WorkflowRunArtifactsResponse {
+  artifacts: Array<
+    {
+      id: number;
+      name: string;
+    }
+  >
+}
+
+/**
+ * Lists the workflow run artifacts for the given workflow run ID.
+ * @returns An object containing the list of artifacts.
+ */
+async function listWorkflowRunArtifacts(
+  credentials: Credentials,
+  owner: string,
+  repo: string,
+  workflowRunId: number
+): Promise<WorkflowRunArtifactsResponse> {
+  const octokit = await credentials.getOctokit();
+  const response = await octokit.rest.actions.listWorkflowRunArtifacts({
+    owner,
+    repo,
+    run_id: workflowRunId,
+  });
+
+  return response.data;
+}
+
+/**
+ * @returns The artifact ID corresponding to the given artifact name.
+ */
+function getArtifactIDfromName(artifactName: string, workflowRunArtifactsResponse: WorkflowRunArtifactsResponse): number | undefined {
+  for (const artifact of workflowRunArtifactsResponse.artifacts) {
```

Comment:
Thanks! That works (10/10 browser coding 😄 )

> Since it becomes a two-liner with the last line figuring out what to return, you could also get rid of the function (but that's all a matter of taste).

I'll keep the function for now, in case we want to re-use it for other artifacts in future 📦